src/rascal/structure_managers/property_block_sparse.hh

```diff
@@ -738,29 +932,126 @@ namespace rascal {
       // You are trying to access a property that does not exist at this depth
       // in the adaptor stack.
       assert(static_cast<int>(CallerLayer) >= this->get_property_layer());
-      return this->values[id.get_cluster_index(this->get_property_layer())]
+      return this->maps[id.get_cluster_index(this->get_property_layer())]
           .get_keys();
     }
 
+    /**
+     * Get a dense feature matrix Ncenter x Nfeatures only if
+     * `this->are_keys_uniform == true`. It only reinterprets the flat data
+     * storage object since all cluster entries have the same number of keys,
+     * i.e. each rows has the same size and has the same key ordering.
+     */
+    MatrixMapConst_Ref_t get_raw_data_view() const {
+      // should not be called if the keys are not uniform
+      assert(this->are_keys_uniform() == true);
```

Comment:
Should this be an exception instead of an assert? `assert` are usually removed in 'release' mode

Reply:
Do you think its not sufficient to check only in debug ? If it is not true the results will be messed up so probably seg fault and to one will go to the debug compilation and get the assert. This is supposed to be used only in few places inside the class not anywhere else.

Reply:
The problem is that you don't have the guarantee of a segfault, and this means that one can get messed up results, think they are reasonable and go on publishing them. So yes, I would rather check more than less.

Reply:
I understand your concern but I feel that this is a bit superfluous since wherever it is used, the flag are_keys_uniform() has already been checked so the scenario you describe is not possible at the moment.

Reply:
>  at the moment.

that's the point: assertions/checks are here to prevent future changes from breaking everything.